File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/JavaScript/_004)_Reverse_Array.js
function reverse(arr) {
    let low = 0;
    let high = arr.length - 1;
    while (low < high) {
        let temp = arr[low];
        arr[low] = arr[high];
        arr[high] = temp;

        low++;
        high--;
    }
}

let arr = [1, 2, 3, 4, 5];

console.log(`Befor Reverse -> ${arr}`);
reverse(arr);
console.log(`After Reverse -> ${arr}`);

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_004)_Reverse_Array.dart
void reverse(arr) {
  var low = 0;
  var high = arr.length - 1;

  while (low < high) {
    var temp = arr[low];
    arr[low] = arr[high];
    arr[high] = temp;

    low++;
    high--;
  }
}

void main() {
  var arr = [1, 2, 3, 4, 5];
  print('Before Reverse: ${arr}');
  reverse(arr);
  print('After Reverse: ${arr}');
}
